361791: x86_64 RHEL4 slapd memory leak

Source: upstream, cvs diff -u -r 1.52 -r 1.53 back-bdb/id2entry.c 
Reviewed-by: Jan Safranek <jsafrane@redhat.com>

Index: back-bdb/id2entry.c
===================================================================
RCS file: /repo/OpenLDAP/pkg/ldap/servers/slapd/back-bdb/id2entry.c,v
retrieving revision 1.52
retrieving revision 1.53
diff -u -r1.52 -r1.53
--- servers/slapd/back-bdb/id2entry.c	6 Mar 2004 18:15:39 -0000	1.52
+++ servers/slapd/back-bdb/id2entry.c	4 Jul 2004 02:56:36 -0000	1.53
@@ -202,6 +202,10 @@
 			SLAP_TRUNCATE_MODE, SLAP_UNDEFINED_MODE */
  
 	if ( slapMode == SLAP_SERVER_MODE ) {
+		/* If not in our cache, just free it */
+		if ( !e->e_private ) {
+			return bdb_entry_return( e );
+		}
 		/* free entry and reader or writer lock */
 		if ( o ) {
 			boi = (struct bdb_op_info *)o->o_private;
